packages/transactions/src/authorization.ts

```diff
@@ -120,7 +120,14 @@ export function createMultiSigSpendingCondition(
   nonce: IntegerType,
   fee: IntegerType
 ): MultiSigSpendingCondition {
-  const stacksPublicKeys = pubKeys.map(createStacksPublicKey);
+  const publicKeys = pubKeys.slice();
+
+  // sort public keys (only for newer non-sequential multi-sig for backwards compatibility)
```

Comment:
This should be handled the same way in all multisig transaction types: We *recommend* that the user sort their pubkeys when creating, but we can't force it because users may have funds in accounts that don't use the recommended ordering

You could add a `boolean` argument, or create another version of this function, `createMultiSigSpendingConditionSorted`, if you want to provide sorting functionality to the user. But you can't just assume the user wants it

Reply:
Got it. I would like to be more opinionated about it, so maybe sorting can be the default and we add a `disablePublicKeySorting` flag. But yeah, this should be a part of all the helpers then as well.. Will change 👍

Reply:
Or maybe not, 🤔 technically the developer can still update and replace conditions/modes/etc. on any transaction fairly easily in Stacks.js (assuming they know what they're doing). Similar to how we don't really _expose_ p2wsh multi-sig, we can "hide" the unsorted variant a bit more as well 🤷🏻‍♀️ trying to find a balance for users to be able to get less wrong by accident.

Reply:
Even with sequential multisig transactions, it was customary to sort the pubkeys before creating a transaction, but it was never enforced. So new multisig types should not handle this differently

One thing you could do, which would work for both types and would be backwards compatible, is to add an optional `address` parameter. Users would be encouraged to supply the address of the multisig account, and you could check if the pubkeys generate that address, first in the order given, then sort them and try in that order. If it doesn't match in either order, return an error

Reply:
Lovely approach!!! I really like this more than another config flag!! 👏

Reply:
Just thinking about next breaking release, I would like to change the default behavior to "sorted" to guide the ecosystem to the preferred way
.